components/moderation-panel/ModEditSubmissionForm.vue

```diff
@@ -1199,66 +548,76 @@ async function submitUpdatedSubmission(e: Event) {
     if (submissionResult) initializeSubmissionFormValues(submissionResult)
     toast.success(t('modSubmissionForm.successMessageUpdated'))
-    if (moderationSubmissionStore.updatingSubmissionFromTopBar) {
-        router.push('/moderation')
+    if (moderationSubmissionStore.updatingSubmissionFromTopBarAndExiting) {
         // reset all modal refs to prevent unintended side effects
-        resetModalRefs()
+        router.push('/moderation')
+        return
+    }
+
+    if (moderationSubmissionStore.updatingSubmissionFromTopBar) {
+        await resetModalRefs()
     }
 }
 
 async function submitCompletedForm(e: Event) {
     // stop the form submitting before we validate
     e.preventDefault()
 
+    const selectedSubmissionData = moderationSubmissionStore.selectedSubmissionData
     //This prevents submission of an already approved submission the backend does this but as an extra visual and check
-    if (submissionFormFields.isApproved) {
+    if (selectedSubmissionData && selectedSubmissionData.isApproved) {
         toast.info(t('modSubmissionForm.infoMessageAlreadyApproved'))
+        await resetModalRefs()
         return
     }
 
-    if (!submissionFormFields.isUnderReview) {
+    if (selectedSubmissionData && !selectedSubmissionData.isUnderReview) {
         toast.info(t('modSubmissionForm.infoMessageUpdateNeeded'))
+        await resetModalRefs()
         return
     }
 
     const isValidFacility = validateFacilityFields()
     const isValidHealthcareProfessional = validateHealthcareProfessionalFields()
 
     //This shows a toast and returns if the facility fields arent valid
-    if (!isValidFacility) {
+    if (!isValidFacility && !currentFacilityRelations.value.length) {
         toast.error(t('modSubmissionForm.errorMessageFacilityInputsInvalid'))
+        await resetModalRefs()
         return
     }
 
-    if (!isValidHealthcareProfessional) {
+    if (!isValidHealthcareProfessional && !currentExistingHealthcareProfessionals.value.length) {
         toast.error(t('modSubmissionForm.errorMessageHealthcareInputsInvalid'))
+        await resetModalRefs()
         return
     }
 
     try {
         const result = await moderationSubmissionStore.approveSubmission()
-        console.log(result)
         if (result?.errors?.length) {
             handleServerErrorMessaging(result.errors, toast, t)
+            await resetModalRefs()
             return
         }
-        modalStore.hideModal()
+        await resetModalRefs()
         toast.success(t('modSubmissionForm.successMessageApproved'))
-        // reset all modal refs to prevent unintended side effects
-        resetModalRefs()
-        router.push('/moderation')
+        await router.push('/moderation')
     } catch {
         toast.error(t('modSubmissionForm.errorMessageCompletedForm'))
+        await resetModalRefs()
     }
 }
 
 const syntheticEvent = new Event('submit', { bubbles: false, cancelable: true })
 
 const resetModalRefs = async () => {
```

Comment:
instead of calling this in every function, I think we can use a watch event or move it to be resused somehow (haven't dug that far yet)